server/channels/app/webhook_test.go

```diff
@@ -381,6 +384,53 @@ Date:   Thu Mar 1 19:46:48 2018 +0300
 	})
 }
 
+func TestCreateWebhookPostWithPriority(t *testing.T) {
+	testCluster := &testlib.FakeClusterInterface{}
+	th := SetupWithClusterMock(t, testCluster).InitBasic()
+	defer th.TearDown()
+
+	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableIncomingWebhooks = true })
+
+	hook, err := th.App.CreateIncomingWebhookForChannel(th.BasicUser.Id, th.BasicChannel, &model.IncomingWebhook{ChannelId: th.BasicChannel.Id})
+	require.Nil(t, err)
+	defer th.App.DeleteIncomingWebhook(hook.Id)
+
+	testConditions := []model.PostPriority{
```

Comment:
@fmartingr - I added a few basic tests here to cover the three primary ways this would be used. I can extend out to more conditions if you think worth it.

Reply:
Yeah that's more than enough. In my head I only want to be sure that all values passed to it are set, checking non-defaults to be sure that they set correctly. 👌